src/main/java/com/deblock/jsondiff/matcher/StrictJsonObjectPartialMatcher.java

```diff
@@ -0,0 +1,40 @@
+package com.deblock.jsondiff.matcher;
+
+import com.deblock.jsondiff.diff.JsonDiff;
+import com.deblock.jsondiff.diff.JsonObjectDiff;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+
+public class StrictJsonObjectPartialMatcher implements PartialJsonMatcher<ObjectNode> {
+
+    @Override
+    public JsonDiff jsonDiff(Path path, ObjectNode expectedJson, ObjectNode receivedJson, JsonMatcher jsonMatcher) {
+        final var jsonDiff = new JsonObjectDiff(path);
+
+        expectedJson.fields()
```

Comment:
Maybe that to improve performance and avoir to perform two loop over all attributes we can do something like that: 

```
final var receviedJsonFields = new HashSet(receivedJson.fields())
// for earch expectedJson.fields
// delete the field on receviedJsonFields

// for earch receviedJsonFields 
// add jsonDiff.addUnexpectedProperty
```

I'm not sur of the cost of delete operation on a Set

Reply:
I'm picking this up now.

Reply:
This MR is not yet ready, there are still edge-cases where I'm running into empty diffs. I'm working on it in another codebase, when I get it fixed there, I'll update this MR.